// Vath/tests/_v_add.test.js
// Loading of the VatomCannon for LowLevel_Vatom_Testing-[per module]
// Hopefully gonna save bunch of time figuring out shitload of code... haha :D 
const VatomCannon = require('vatom_cannon')
//----------------------------------------------------------------


// Getting the actuall Vath to use in the tests
const vath = require('vath')
//----------------------------------------------

// This is the test settings VatomCannon is going to use
const v_add_tests = {
	printAll: false,
	name: "v_add",  // Some name for output to have idea what's it testing.
	hint: "Add more man! Cover The Base Hill With Bodies If Needed...buahaha", // Limited Output to 32chars for the benefit of good look. :D 
	targetList: [{
		desc: "No input values [ ??? + ??? => ??? ]",
		pass: false,
		tested: false,
		test() {
			return v_add();
		},
		expected: "ERROR: Missing Params vath.v_add(x,y)"
	},
	{
		desc: "Missing Second Input Variable [ 11 + missing => ??? ]",
		pass: false,
		tested: false,
		test() {
			return v_add(11);
		},
		expected: "ERROR: Missing Params vath.v_add(x,y)"
	},
	{
		desc: "Input Variables Are OK [ 22 + 55 => 77]",
		pass: false,
		tested: false,
		test() {
			return v_add(22, 55);
		},
		expected: 77
	},
	{
		desc: "[TestFailing_DEMO] Input Variables Are OK {but expected is set to 771 \n| while we are doing same thing [ 22 + 55 = 77(1 <- example of human error ] as previous one only with mistake}",
		pass: false,
		tested: false,
		test() {
			return v_add(22, 55);
		},
		expected: 771,
		ignored: true
	},
	{
		desc: "Lets send it some text",
		pass: false,
		tested: false,
		test() {
			return v_add("22", 55);
		},
		expected: '2255'
	},
	{
		desc: "Lets send it some text",
		pass: false,
		tested: false,
		test() {
			return v_add("22", 55);
		},
		expected: "2255"
	}

	]
};
//------------------------------------------------------


// Just give it the list to go through
// Setup of constant for test result export
const v_test_res_add = VatomCannon(v_add_tests);
//------------------------------------------

// and the export of it.
module.exports = v_test_res_add;
//------------------------------
